### PR TITLE
sbt: 1.2.8 -> 1.3.2

### PR DESCRIPTION
**Pull request by regadas**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/development/tools/build-managers/sbt/default.nix [hunk 1]
```diff
@@ -2,15 +2,15 @@
...
+  version = "1.3.2";
 
   src = fetchurl {
     urls = [
       "https://dl.bintray.com/sbt/native-packages/sbt/${version}/${pname}-${version}.tgz"
       "https://github.com/sbt/sbt/releases/download/v${version}/sbt-${version}.tgz"
       "https://cocl.us/sbt-${version}.tgz"
     ];
-    sha256 = "0n7yghnb1q3lyjpv721znvslk5lwib7y84mxwz66yv8p84jj3fcv";
+    sha256 = "1ncjc4yrnhp2kc5xw3ismnyx5n71541760rzs81hxi8x8vi36x5b";
```
Comment (Mic92) [inline pkgs/development/tools/build-managers/sbt/default.nix]: I get a different hash:

```
hash mismatch in fixed-output derivation '/nix/store/v8f3lbkzkr5gjkm6m794xhzrqfnnpmiz-sbt-1.3.2.tgz':
  wanted: sha256:1ncjc4yrnhp2kc5xw3ismnyx5n71541760rzs81hxi8x8vi36x5b
  got:    sha256:0vrj9wlw5kf6w7nzhf58dyik0n7rnc9aivjpsz85m299j4wyz37d
```

Comment (regadas) [reply pkgs/development/tools/build-managers/sbt/default.nix]: @Mic92 interesting ... not sure if this is the correct way to get the hash:
```
nix-prefetch-url --unpack https://github.com/sbt/sbt/releases/download/v1.3.2/sbt-1.3.2.tgz
unpacking...
[49.4 MiB DL]
path is '/nix/store/2mvik6xvnas8s3i8c78lmlljq15a2g3l-sbt-1.3.2.tgz'
1ncjc4yrnhp2kc5xw3ismnyx5n71541760rzs81hxi8x8vi36x5b
```

Comment (Mic92) [reply pkgs/development/tools/build-managers/sbt/default.nix]: @regadas is it possible that the bintray archive is different from the github one?

Comment (regadas) [reply pkgs/development/tools/build-managers/sbt/default.nix]: Hi @Mic92 it's definitely possible; tbh only the github link is being updated with the latest releases the others will just 404; I think we should just keep the github one, wdyt?

Comment (Mic92) [reply pkgs/development/tools/build-managers/sbt/default.nix]: @regadas yes. drop the other ones.

File: pkgs/development/tools/build-managers/sbt/default.nix [hunk 2]
```diff
@@ -2,15 +2,14 @@
 
 stdenv.mkDerivation rec {
   pname = "sbt";
-  version = "1.2.8";
+  version = "1.3.2";
 
   src = fetchurl {
     urls = [
-      "https://dl.bintray.com/sbt/native-packages/sbt/${version}/${pname}-${version}.tgz"
+      "https://piccolo.link/sbt-${version}.tgz",
```
Comment (regadas) [inline pkgs/development/tools/build-managers/sbt/default.nix]: This is being used in [https://www.scala-sbt.org/download.html](https://www.scala-sbt.org/download.html)

File: pkgs/development/tools/build-managers/sbt/default.nix (hunk 1 above)
Comment (regadas) [reply pkgs/development/tools/build-managers/sbt/default.nix]: updated ... and they have the same hash using `nix-prefetch-url`

**Comment by regadas** [comment]
@Mic92 should be good now!

**Comment by Mic92** [comment]
@GrahamcOfBorg build sbt

File: pkgs/development/tools/build-managers/sbt/default.nix (hunk 1 above)
Comment (Mic92) [reply pkgs/development/tools/build-managers/sbt/default.nix]: The checksum is still incorrect.

Comment (regadas) [reply pkgs/development/tools/build-managers/sbt/default.nix]: @Mic92 hmm,

```
nix-prefetch-url --unpack https://piccolo.link/sbt-1.3.2.tgz    unpacking...
path is '/nix/store/2mvik6xvnas8s3i8c78lmlljq15a2g3l-sbt-1.3.2.tgz'
1ncjc4yrnhp2kc5xw3ismnyx5n71541760rzs81hxi8x8vi36x5b

nix-prefetch-url --unpack https://github.com/sbt/sbt/releases/download/v1.3.2/sbt-1.3.2.tgz      
unpacking...
path is '/nix/store/2mvik6xvnas8s3i8c78lmlljq15a2g3l-sbt-1.3.2.tgz'
1ncjc4yrnhp2kc5xw3ismnyx5n71541760rzs81hxi8x8vi36x5b
```

Comment (regadas) [reply pkgs/development/tools/build-managers/sbt/default.nix]: I'll have a better look at this later ... looks like smth else might be broken?

Comment (Mic92) [reply pkgs/development/tools/build-managers/sbt/default.nix]: Ofborg also shows a different checksum: https://logs.nix.ci/?key=nixos/nixpkgs.69327&attempt_id=91c42493-27e4-4271-8664-3e6aa9c037fd

Can you build the package manually?

Comment (Mic92) [reply pkgs/development/tools/build-managers/sbt/default.nix]: Ah. You should not use `--unpack` when `fetchurl` is used. Fetchurl is calculating the checksum of the tarball and not of the directory content.

Comment (regadas) [reply pkgs/development/tools/build-managers/sbt/default.nix]: Ahh ok! i thought it would check the content ... good to know 👍

**Comment by regadas** [comment]
@GrahamcOfBorg build sbt

**Comment by Mic92** [comment]
@GrahamcOfBorg build sbt

**Pull request merged**